core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,153 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a lessons in learner dashboard
+ */
+
+import {Component, Input, OnInit} from '@angular/core';
+import {downgradeComponent} from '@angular/upgrade/static';
+import {AppConstants} from 'app.constants';
+import {AssetsBackendApiService} from 'services/assets-backend-api.service';
+import {UrlInterpolationService} from 'domain/utilities/url-interpolation.service';
+import {CollectionSummary} from 'domain/collection/collection-summary.model';
+import {LearnerExplorationSummary} from 'domain/summary/learner-exploration-summary.model';
+import {StorySummary} from 'domain/story/story-summary.model';
+import {StoryNode} from 'domain/story/story-node.model';
+
+interface LessonUrl {
+  classroom: string | undefined;
+  topicFragment: string | undefined;
+  story: string;
+  currentStory: StoryNode;
+}
+
+interface ThumbnailUrl {
+  filename: string;
+  id: string;
+}
+@Component({
+  selector: 'lesson-card',
+  templateUrl: './lesson-card.component.html',
+})
+export class LessonCardComponent implements OnInit {
+  @Input() story!: StorySummary | LearnerExplorationSummary | CollectionSummary;
+  @Input() topic!: string;
+
+  desc!: string;
+  imgColor!: string;
+  imgUrl!: string;
+  lessonUrl!: string;
+  progress!: number;
+  title!: string;
+  lessonTopic!: string;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      let completedStories = this.story.getCompletedNodeTitles().length;
+
+      this.desc = this.story.getTitle();
+      this.imgColor = this.story.getThumbnailBgColor();
+      this.imgUrl = this.getStorySummaryThumbnailUrl({
+        filename: this.story.getThumbnailFilename(),
+        id: this.story.getId(),
+      });
+
+      let nextStory =
+        completedStories === this.story.getAllNodes().length
+          ? completedStories - 1
+          : completedStories;
+      /* Last completed story index works because if 1 is completed, 1 index is 2nd item */
+      let lessonArgs = {
+        classroom: this.story.getClassroomUrlFragment(),
+        topicFragment: this.story.getTopicUrlFragment(),
+        currentStory: this.story.getAllNodes()[nextStory],
+        story: this.story.getUrlFragment(),
+      };
+      this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+      this.title = `Chapter ${nextStory + 1}: ${this.story.getNodeTitles()[nextStory]}`;
+      this.progress = Math.floor(
+        (completedStories / this.story.getNodeTitles().length) * 100
+      );
+      this.lessonTopic = this.topic;
+    } else {
+      /* Implementation for explorations & collections */
```

Comment:
If you want to add a comment try first refactor the code so that the comment becomes redundant. Here, this is a good candidate for a function, name it properly and everyone will understand what are you doing here

Reply:
This is not fixed

Reply:
I believe you might be looking at the previous commits. Here's the [most recent commits](https://github.com/oppia/oppia/pull/20185/files/fda6aec2ba8b58c3c9fe5b49d61555312c349331..7ef036875e749c83eb794578d261710132b96f73)

Reply:
Nope. You deleted the comment, but you added it in the first place because the code is not too clear. 

"Here, this is a good candidate for a function, name it properly and everyone will understand what are you doing here"

Reply:
Ah, I see.